docs/src/test/scala/sample/scaladsl/ConsumerExample.scala

```diff
@@ -341,3 +341,29 @@ object ExternallyControlledKafkaConsumer extends ConsumerExample {
   }
 }
 
+//Restart Kafka consumer
+object RestartKafkaConsumeExample extends ConsumerExample {
+  def main(args: Array[String]): Unit = {
+    // #restartConsumer
+    val runnableGraph: RunnableGraph[Consumer.Control] =
+      Consumer.committableSource(consumerSettings, Subscriptions.topics("topic1"))
+        .map(msg => { println(msg.record.value()); msg.committableOffset })
+        .map(_.commitScaladsl())
+        .toMat(Sink.ignore)(Keep.left)
+    val control1: Consumer.Control = runnableGraph.run()
+    val shutdownFuture = control1.shutdown()
+
+    shutdownFuture onComplete {
+      case Success(Done) =>
+        println("Stream was shutdown")
+
+        // Start the stream again
+        val control2: Consumer.Control = runnableGraph.run()
```

Comment:
I am not sure if this is something we'd recommend. How would you return this `control2` object for further error handling? Maybe it would be better to manage the graph in a an actor and make it self-restart when the stream gets shut down or fails?